### ISSUE TITLE
mobile onboarding: mega issue

### ISSUE THREAD

**Issue by zalberico** [tloncorp/landscape-apps#2023]
I went through the current onboarding flow with @emceetee sharing my phone screen. I'm writing up the details here and can spin out bits into separate issues where it makes sense. Some of this touches hosting, some of it touches homestead. For context: @lukebuehler @lukestiles @jamesacklin @arthyn 

- Mariah first emailed me a link to sign up (from the Tlon public sign up dashboard page flow)
- I clicked the link, there's too much text on this page we can reduce this down to just the call to action.
- On the Choose your Urbit ID screen, the text above and below the white boxes is hard to read (black text on dark background)
- ![F56FDB0E-A495-47DF-AA18-027FF9E3D3D3](https://user-images.githubusercontent.com/441466/221969044-05320881-e0de-493d-908f-5b98a91fe98d.png)
- The ID selector doesn't allow you to swipe left or right, you can click the sigils or the little arrows, attempting swipe move the entire page.
- Once signed up you go to the hosting dashboard page, this page is fairly busy and has a blue link to take you to pending group invites. This takes you to the urbit login asking for access key. User's never copy the access key from the dashboard page. We need to skip as much of this as we can and just drop users into their urbit. This also triggered an email to the user from hosting (which is good).
- Once logged in (after copying their access key) the user sees suggested groups, but they take 30s to load the proper name and image, notably my group also had no pending invite to tlon public (but with these default groups that doesn't matter as much). Urbit Community should be a default group listed here.
- ![30AE13FC-B807-458B-A6D7-6A01DA0ED930](https://user-images.githubusercontent.com/441466/221969609-f0ea860f-2643-4c4d-b564-5e4afa3837fa.png)
- When you join a group you are dropped into a blank activity screen. This is a bad default - we should put users on the channel page. I'm not sure this notification tab makes sense here at all.
- ![A53E00F2-EBE2-4DE9-A2F4-AA456890D89F](https://user-images.githubusercontent.com/441466/221969730-564098d9-2657-4787-9cf9-e027b828d0b0.png)
- When you ask a user to create a group they see this screen:
- ![6073BF88-EF55-4211-992D-D914124CD7EA_1_101_o](https://user-images.githubusercontent.com/441466/221971896-c243a207-b088-4e3a-a992-353502ef31cd.jpeg)
- There's lots of problematic stuff here, the hex color selector, the paste/image, switching back and forth between those, the behavior without s3 support etc. This needs attention.
- The group creation flow asks the user to invite patps except they're new and don't know any. This really needs to be inviting emails which leverage lure to bring people into urbit via hosting.
- The user then creates the group and gets dropped into another blank notifications screen, there's no path to creating a channel. We should have creating a chat channel as part of the group creation flow (or should just create one by default).
- Channel creation should probably get rid of the concept of default channels and have channels always be default (and added to existing users). Hiding them in the all channels section makes them hard to discover.
- When you do create a channel, after you create it you're dropped again back into the blank notification screen.
- It's *impossible* currently to get from groups to talk on mobile. There's no way to get to the app tile page from the UI. Since hosting drops the user right into groups when the click the blue link and sign in - this is an issue.
- Lastly there also the (known) issue with profiles
- - ![96154264-88C7-4E85-94D9-0B63F4119000_1_201_a](https://user-images.githubusercontent.com/441466/221969845-238b0a64-c960-4792-8569-36b819715985.jpeg)
- We should probably have the small talk install link right on the suggested groups page.

That's the main stuff I noticed on my initial 'self-onboarding' flow. We should also have Urbit Community in the default group list, but stuff generally worked beyond that - if we can fix a lot of the way these flows work it should make things a lot better. On the hosting side, password reset flow should also be possible from accounts page (not only from sign in page). Also after resetting password I shouldn't have to log in again.

**Comment by jamesacklin** [comment]
Thank you, Zach.

I've consolidated all the feedback from launching Landscape up to (but not including) creating a new group. I'm treating this as a single flow: landing in Groups, joining a group from the suggested list, and getting dropped into the group.


![Image](https://user-images.githubusercontent.com/748181/221996265-6e4b16ac-8f93-4b19-8959-1cd3b2f8f402.png)

The flow should be self-evident, but we need to do some work to complete this chapter. From left to right, here are the corresponding issues.

- Faking the list of suggested groups out to appear to load faster: #2027
- Dropping new joiners into /channellist rather than an empty /activity: #2028
- Auto-joining all channels: #1997

I will similarly treat the new group creation flow. Plugging Small Talk could be interesting.

**Comment by lukebuehler** [comment]
Great writeup! Some of the onboarding issues are being fixed by @tomholford right now. See https://github.com/tloncorp/ylem/issues/826

**Comment by arthyn** [comment]
Btw I think UC was avoided because joining it right now is pretty harrowing and looks bad. Need solid state subscriptions to improve this or some other measure to snapshot state that we're syncing.

As far as inviting people goes this seems pretty tricky to get right. We could mix patps and email together potentially or maybe have some toggle to switch? Just brainstorming here, but I do think it's important that we include this ability to bring more people in. We should have this capability once lure service is complete right?

I know for sure we've talked about this step in the flow of group creation being problematic to begin with. Maybe a better thing to do is kill the step, have some other prompts in the new group to invite people, and that can have flows for both patp and email.

**Comment by jamesacklin** [comment]
In my opinion, deferring (A) further customization and (B) inviting more ships to _after_ group creation will go a long way here. Filling out a multi-part form and getting dropped into an empty group isn't great.

**Comment by zalberico** [comment]
Ack on delaying customization and inviting more ships to post creation (though channel creation maybe?). For UC we should include and handle delayed loading state.

**Comment by jamesacklin** [comment]
@zalberico Closing this mega-issue. We addressed most of the post-Groups-launch issues (or at least account for them) and @urcades is working on the pre-Groups-launch flow. 

> When you join a group you are dropped into a blank activity screen. This is a bad default - we should put users on the channel page. I'm not sure this notification tab makes sense here at all.

Fixed. New users are redirected to the Find Groups page.

> When you ask a user to create a group they see this screen: (create group dialog) There's lots of problematic stuff here, the hex color selector, the paste/image, switching back and forth between those, the behavior without s3 support etc. This needs attention.

Opening a new issue.

> The group creation flow asks the user to invite patps except they're new and don't know any. This really needs to be inviting emails which leverage lure to bring people into urbit via hosting.

Opening a new issue.

> The user then creates the group and gets dropped into another blank notifications screen, there's no path to creating a channel. We should have creating a chat channel as part of the group creation flow (or should just create one by default).

Fixed. We drop you directly into the channel manager.

> Channel creation should probably get rid of the concept of default channels and have channels always be default (and added to existing users). Hiding them in the all channels section makes them hard to discover.

Fixed. Everyone joins every channel.

> When you do create a channel, after you create it you're dropped again back into the blank notification screen.

Fixed. We keep you in the channel manager.

> It's impossible currently to get from groups to talk on mobile. There's no way to get to the app tile page from the UI. Since hosting drops the user right into groups when the click the blue link and sign in - this is an issue.

For further consideration by the design team in our "system chrome" efforts.

> Lastly there also the (known) issue with profiles

Fixed. Profiles work and we haven't taken down the network with subscription load (yet), so we'll cautiously open up more avenues to initiate the subscription.

> We should probably have the small talk install link right on the suggested groups page.

Good idea, opening another issue.
